### ISSUE TITLE
Add pkg mgmt vgo  and research about implementation/use in go-ceph

### ISSUE THREAD

**Issue by MalloZup** [ceph/go-ceph#70]
Hi all, if you are ok i could spend a PR on this.

```golang
make test
go test -v ./...
# github.com/ceph/go-ceph
package_test.go:4:2: cannot find package "github.com/stretchr/testify/assert" in any of:
	/usr/lib64/go/1.9/src/github.com/stretchr/testify/assert (from $GOROOT)
	/home/dmaiocchi/go/src/github.com/stretchr/testify/assert (from $GOPATH)
FAIL	github.com/ceph/go-ceph [setup failed]
# github.com/ceph/go-ceph/cephfs
package_test.go:4:2: cannot find package "github.com/stretchr/testify/assert" in any of:
	/usr/lib64/go/1.9/src/github.com/stretchr/testify/assert (from $GOROOT)
	/home/dmaiocchi/go/src/github.com/stretchr/testify/assert (from $GOPATH)
FAIL	github.com/ceph/go-ceph/cephfs [setup failed]
# github.com/ceph/go-ceph/rados
rados/rados_test.go:17:2: cannot find package "github.com/gofrs/uuid" in any of:
	/usr/lib64/go/1.9/src/github.com/gofrs/uuid (from $GOROOT)
	/home/dmaiocchi/go/src/github.com/gofrs/uuid (from $GOPATH)
FAIL	github.com/ceph/go-ceph/rados [setup failed]
# github.com/ceph/go-ceph/rbd
package_test.go:4:2: cannot find package "github.com/stretchr/testify/assert" in any of:
	/usr/lib64/go/1.9/src/github.com/stretchr/testify/assert (from $GOROOT)
	/home/dmaiocchi/go/src/github.com/stretchr/testify/assert (from $GOPATH)
FAIL	github.com/ceph/go-ceph/rbd [setup failed]
```

**Comment by noahdesu** [comment]
Does this offer an advantage of running `go get ./...` to grab the dependencies?

**Comment by MalloZup** [comment]
@noahdesu i  will have a look for `vgo`

**Comment by noahdesu** [comment]
Thanks @MalloZup. I'll also look at vgo and get up to speed on what's happening in that space. I'm all for adopting new tools and techniques for the project. Were you using ceph-go in a way that made compatibility with different version a problem? Maybe we can address a few specific cases without vendoring everything.

**Comment by isi-lincoln** [comment]
I'd offer up a contrary package management tool `dep`: https://github.com/golang/dep

My own experience is that dep is taking over as the go dependency package manager.

**Comment by dswarbrick** [comment]
@isi-lincoln dep has been the preferred dependency management tool for the last few years. I've used it myself, and have nothing against it. But with Go 1.11 support for modules, it is more likely that vgo with modules will be the path forward.

**Comment by MalloZup** [comment]
@isi-lincoln yop agree with @dswarbrick.there is also the announcement upstream in golang

**Comment by noahdesu** [comment]
Thank you everyone for chiming in. I've found myself maintaining `go-ceph`, but I don't use Go or `go-ceph` on a day to day basis :) so this input is really valuable.

It sounds like in the short term everyone can benefit from tagged releases, and I can make that happen. Further, sometime in the future vgo will be used, but there is nothing to do w.r.t. to vgo / dep at this moment?

**Comment by MalloZup** [comment]
@noahdesu  i think yop is not worth for the moment to use `go dep`. At moment you are using mainly the testing library.

I think that the `vgo` is really imminent https://golang.org/dl/ already released the 1.11 so one could already try to do the thing
